exporters/elasticsearch/include/opentelemetry/exporters/elasticsearch/es_log_recordable.h

```diff
@@ -0,0 +1,190 @@
+
+  /**
+   * Set a resource for this log.
+   * @param name the name of the resource
+   * @param value the resource value
+   */
+  void SetResource(nostd::string_view key,
+                   const opentelemetry::common::AttributeValue &value) noexcept override
+  {
+    switch (value.index())
+    {
+      case common::AttributeType::TYPE_BOOL:
+        json_["resource"][key.data()] = opentelemetry::nostd::get<bool>(value) ? true : false;
+        break;
+      case common::AttributeType::TYPE_INT:
+        json_["resource"][key.data()] = opentelemetry::nostd::get<int>(value);
+        break;
+      case common::AttributeType::TYPE_INT64:
+        json_["resource"][key.data()] = opentelemetry::nostd::get<int64_t>(value);
+        break;
+      case common::AttributeType::TYPE_UINT:
+        json_["resource"][key.data()] = opentelemetry::nostd::get<unsigned int>(value);
+        break;
+      case common::AttributeType::TYPE_UINT64:
+        json_["resource"][key.data()] = opentelemetry::nostd::get<uint64_t>(value);
+        break;
+      case common::AttributeType::TYPE_DOUBLE:
+        json_["resource"][key.data()] = opentelemetry::nostd::get<double>(value);
+        break;
+      case common::AttributeType::TYPE_STRING:
+      case common::AttributeType::TYPE_CSTRING:
+        json_["resource"][key.data()] =
+            opentelemetry::nostd::get<opentelemetry::nostd::string_view>(value).data();
+        break;
+      default:
+        break;
+    }
+  }
+
+  /**
+   * Set an attribute of a log.
+   * @param key the key of the attribute
+   * @param value the attribute value
+   */
+  void SetAttribute(nostd::string_view key,
+                    const opentelemetry::common::AttributeValue &value) noexcept override
+  {
+    switch (value.index())
+    {
+      case common::AttributeType::TYPE_BOOL:
+        json_["attributes"][key.data()] = opentelemetry::nostd::get<bool>(value) ? true : false;
+        break;
+      case common::AttributeType::TYPE_INT:
+        json_["attributes"][key.data()] = opentelemetry::nostd::get<int>(value);
+        break;
+      case common::AttributeType::TYPE_INT64:
+        json_["attributes"][key.data()] = opentelemetry::nostd::get<int64_t>(value);
+        break;
+      case common::AttributeType::TYPE_UINT:
+        json_["attributes"][key.data()] = opentelemetry::nostd::get<unsigned int>(value);
+        break;
+      case common::AttributeType::TYPE_UINT64:
+        json_["attributes"][key.data()] = opentelemetry::nostd::get<uint64_t>(value);
+        break;
+      case common::AttributeType::TYPE_DOUBLE:
+        json_["attributes"][key.data()] = opentelemetry::nostd::get<double>(value);
+        break;
+      case common::AttributeType::TYPE_STRING:
+      case common::AttributeType::TYPE_CSTRING:
+        json_["attributes"][key.data()] =
+            opentelemetry::nostd::get<opentelemetry::nostd::string_view>(value).data();
+        break;
+      default:
+        break;
+    }
+  }
```

Comment:
Nit: It seems most of the code is repeated in these two functions `SetAttributes` and `SetResources`. Can we do instead with single function, either through some template function, or pass another argument depicting "attributes" or "resource".

Reply:
Yup good idea, I'll move it over into a function that takes in a string as the parameter.